pennylane/ops/op_math/sprod.py

```diff
@@ -254,9 +253,6 @@ def has_matrix(self):
 
     @staticmethod
     def _matrix(scalar, mat):
-        if qml.math.get_interface(scalar) == "tensorflow":
-            # we must cast ``scalar`` to complex to avoid an error
-            scalar = qml.math.cast_like(scalar, mat)
```

Comment:
this casting is already done in the calling function (scalarsymbolicop.matrix, updated in this PR)